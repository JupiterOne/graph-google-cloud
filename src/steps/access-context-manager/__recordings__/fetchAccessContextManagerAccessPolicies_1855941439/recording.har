{
  "log": {
    "_recordingName": "fetchAccessContextManagerAccessPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 572,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 572,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:23 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.123Z",
        "time": 212,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 212
        }
      },
      {
        "_id": "e7ae7a9d3285d95d4330a7c239aa61b7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.2 gl-node/14.15.4 auth/7.1.0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.2 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "accesscontextmanager.googleapis.com"
            }
          ],
          "headersSize": 581,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "parent",
              "value": "organizations/958457776463"
            }
          ],
          "url": "https://accesscontextmanager.googleapis.com/v1/accessPolicies?parent=organizations%2F958457776463"
        },
        "response": {
          "bodySize": 369,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 369,
            "text": "{\"accessPolicies\":[{\"name\":\"accessPolicies/368605946207\",\"parent\":\"organizations/958457776463\",\"title\":\"default policy\",\"etag\":\"64186ca0c82b01b4\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Wed, 02 Jun 2021 14:22:24 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 488,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-02T14:22:23.374Z",
        "time": 829,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 829
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
