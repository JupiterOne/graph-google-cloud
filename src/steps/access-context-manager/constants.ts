export const STEP_ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES =
  'fetch-access-context-manager-access-policies';
export const STEP_ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS =
  'fetch-access-context-manager-access-levels';
export const STEP_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS =
  'fetch-access-context-manager-service-perimeters';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY = 'AccessPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_POLICY =
  'google_access_context_manager_access_policy';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_ACCESS_LEVEL = 'Ruleset';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_ACCESS_LEVEL =
  'google_access_context_manager_access_level';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER =
  'google_access_context_manager_service_perimeter';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_EGRESS_POLICY =
  'ControlPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_EGRESS_POLICY =
  'google_access_context_manager_service_perimeter_egress_policy';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_INGRESS_POLICY =
  'ControlPolicy';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_INGRESS_POLICY =
  'google_access_context_manager_service_perimeter_ingress_policy';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_API_OPERATION =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_API_OPERATION =
  'google_access_context_manager_service_perimeter_api_operation';

export const ENTITY_CLASS_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_METHOD_SELECTOR =
  'Configuration';
export const ENTITY_TYPE_ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETER_METHOD_SELECTOR =
  'google_access_context_manager_service_perimeter_method_selector';

export const RELATIONSHIP_TYPE_ACCESS_POLICY_HAS_ACCESS_LEVEL =
  'google_access_context_manager_access_policy_has_level';
export const RELATIONSHIP_TYPE_ACCESS_POLICY_HAS_SERVICE_PERIMETER =
  'google_access_context_manager_access_policy_has_service_perimeter';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_PROTECTS_PROJECT =
  'google_access_context_manager_service_perimeter_protects_project';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_LIMITS_API_SERVICES =
  'google_access_context_manager_service_perimeter_limits_api_service';

export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_HAS_EGRESS_POLICY =
  'google_access_context_manager_service_perimeter_has_egress_policy';
export const RELATIONSHIP_TYPE_EGRESS_POLICY_HAS_API_OPERATION =
  'google_access_context_manager_service_perimeter_egress_policy_has_api_operation';
export const RELATIONSHIP_TYPE_API_OPERATION_HAS_METHOD_SELECTOR =
  'google_access_context_manager_service_perimeter_api_operation_has_method_selector';
export const RELATIONSHIP_TYPE_SERVICE_PERIMETER_HAS_INGRESS_POLICY =
  'google_access_context_manager_service_perimeter_has_ingress_policy';
export const RELATIONSHIP_TYPE_INGRESS_POLICY_HAS_API_OPERATION =
  'google_access_context_manager_service_perimeter_ingress_policy_has_api_operation';

export const IngestionSources = {
  ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES: 'acm-access-policies',
  ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS: 'acm-access-levels',
  ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS: 'acm-service-perimeters',
};

export const AccessContextManagerIngestionConfig = {
  [IngestionSources.ACCESS_CONTEXT_MANAGER_ACCESS_POLICIES]: {
    title: 'Google Cloud Access Context Manager Access Policies',
    description: 'Defines secure access boundaries in GCP.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ACCESS_CONTEXT_MANAGER_ACCESS_LEVELS]: {
    title: 'Google Cloud Access Context Manager Access Levels',
    description: 'Manages access via hierarchical levels in GCP.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ACCESS_CONTEXT_MANAGER_SERVICE_PERIMETERS]: {
    title: 'Google Cloud Access Context Manager Service Perimeters',
    description: 'Secures resources within network boundaries.',
    defaultsToDisabled: false,
  },
};
