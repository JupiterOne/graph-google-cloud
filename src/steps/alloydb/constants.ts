export const STEP_ALLOYDB_POSTGRE_SQL_SERVICE = 'create-postgre-sql-service';
export const STEP_ALLOYDB_POSTGRE_SQL_CLUSTER =
  'fetch-alloydb-postgre-sql-cluster';
export const STEP_ALLOYDB_POSTGRE_SQL_INSTANCE =
  'fetch-alloydb-postgre-sql-instance';
export const STEP_ALLOYDB_POSTGRE_SQL_CONNECTION =
  'fetch-alloydb-postgre-sql-connection';
export const STEP_ALLOYDB_POSTGRE_SQL_BACKUP =
  'fetch-alloydb-postgre-sql-backup';

// Entity type
export const ENTITY_TYPE_ALLOYDB_POSTGRE_SQL_SERVICE = 'google_cloud_alloydb';
export const ENTITY_CLASS_ALLOYDB_POSTGRE_SQL_SERVICE = ['Service'];

export const ENTITY_TYPE_POSTGRE_SQL_ALLOYDB_CLUSTER =
  'google_cloud_alloydb_cluster';
export const ENTITY_CLASS_POSTGRE_SQL_CLUSTER = [
  'Database',
  'DataStore',
  'Cluster',
];

export const ENTITY_TYPE_POSTGRE_SQL_ALLOYDB_INSTANCE =
  'google_cloud_alloydb_instance';
export const ENTITY_CLASS_POSTGRE_SQL_INSTANCE = [
  'Database',
  'DataStore',
  'Host',
];

export const ENTITY_TYPE_POSTGRE_SQL_CONNECTION =
  'google_cloud_alloydb_connection';
export const ENTITY_CLASS_POSTGRE_SQL_CONNECTION = ['Network'];

export const ENTITY_TYPE_POSTGRE_SQL_BACKUP = 'google_cloud_alloydb_backup';
export const ENTITY_CLASS_POSTGRE_SQL_BACKUP = ['Backup'];

// Relationships
export const STEP_PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP =
  'build-project-alloydb-service-relationship';
export const STEP_PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP =
  'build-project-has-alloydb-cluster-relationship';

export const STEP_ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP =
  'build-alloydb-instance-uses-cluster-relationship';

export const STEP_ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP =
  'build-alloydb-cluster-has-backup-relationship';

export const STEP_ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP =
  'build-alloydb-instance-has-connection-relationship';

export const STEP_ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP =
  'build-alloydb-cluster-uses-kms-key-relationship';

export const STEP_USER_ASSIGNED_ALLOYDB_CLUSTER_RELATIONSHIP =
  'build-user-asssigned-alloydb-cluster-relatipnship';

export const RELATIONSHIP_TYPE_PROJECT_HAS_ALLOYDB_SERVICE =
  'google_cloud_project_has_alloydb';
export const RELATIONSHIP_TYPE_PROJECT_HAS_ALLOYDB_CLUSTER =
  'google_cloud_project_has_alloydb_cluster';
export const RELATIONSHIP_TYPE_ALLOYDB_INSTANCE_USES_CLUSTER =
  'google_cloud_alloydb_instance_uses_cluster';
export const RELATIONSHIP_TYPE_ALLOYDB_CLUSTER_HAS_BACKUP =
  'google_cloud_alloydb_cluster_has_backup';
export const RELATIONSHIP_TYPE_STEP_ALLOYDB_INSTANCE_HAS_CONNECTION_BACKUP =
  'google_cloud_alloydb_instance_has_connection';
export const RELATIONSHIP_TYPE_ALLOYDB_CLUSTER_USES_KMS_KEY =
  'google_cloud_alloydb_cluster_uses_kms_crypto_key';
export const RELATIONSHIP_TYPE_USER_ASSIGNED_ALLOYDB_CLUSTER =
  'google_user_assigned_cloud_alloydb_cluster';

// IngestionSources
export const IngestionSources = {
  ALLOYDB_POSTGRE_SQL_SERVICE: 'postgre-sql-service',
  ALLOYDB_POSTGRE_SQL_CLUSTER: 'alloydb-postgre-sql-cluster',
  ALLOYDB_POSTGRE_SQL_INSTANCE: 'alloydb-postgre-sql-instance',
  ALLOYDB_POSTGRE_SQL_CONNECTION: 'alloydb-postgre-sql-connection',
  ALLOYDB_POSTGRE_SQL_BACKUP: 'alloydb-postgre-sql-backup',
  PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP:
    'project-has-alloydb-service-relationship',
  PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP:
    'project-has-alloydb-cluster-relationship',
  ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP:
    'alloydb-instance-uses-cluster-relationship',
  ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP:
    'alloydb-cluster-has-backup-relationship',
  ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP:
    'alloydb-instance-has-connection-relationship',
  ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP:
    'alloydb-cluster-uses-kms-key-relationship',
  GOOGLE_USER_ASSIGNED_ALLOYDB_CLUSTER_RELATIONSHIP:
    'google-user-assigned-alloydb-cluster-relationship',
};

// IngestionSources Configs
export const AlloyDBIngestionConfig = {
  [IngestionSources.ALLOYDB_POSTGRE_SQL_SERVICE]: {
    title: 'AlloyDB for Postgre SQL service',
    description: 'AlloyDB for Postgre SQL service.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_CLUSTER]: {
    title: 'AlloyDB for Postgre SQL Cluster',
    description: 'AlloyDB for Postgre SQL Cluster.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_INSTANCE]: {
    title: 'AlloyDB for Postgre SQL Instance',
    description: 'AlloyDB for Postgre SQL Instance.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_POSTGRE_SQL_CONNECTION]: {
    title: 'AlloyDB for Postgre SQL Connection',
    description: 'AlloyDB for Postgre SQL Connection.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ALLOYDB_SERVICE_RELATIONSHIP]: {
    title: 'Project AlloyDB Service Has Relationship',
    description: 'Build Project AlloyDB Service Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_ALLOYDB_CLUSTER_RELATIONSHIP]: {
    title: 'Project AlloyDB Cluster Has Relationship',
    description: 'Build Project AlloyDB Cluster Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_INSTANCE_USES_CLUSTER_RELATIONSHIP]: {
    title: 'AlloyDB Instance Cluster Uses Relationship',
    description: 'Build AlloyDB Instance Cluster Uses Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_CLUSTER_HAS_BACKUP_RELATIONSHIP]: {
    title: 'AlloyDB Instance Backup Has Relationship',
    description: 'Build AlloyDB Backup Cluster Has  Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_INSTANCE_HAS_CONNECTION_RELATIONSHIP]: {
    title: 'AlloyDB Instance Connection Has  Relationship',
    description: 'Build AlloyDB Instance Connection Has Relationship.',
    defaultsToDisabled: false,
  },
  [IngestionSources.ALLOYDB_CLUSTER_USES_KMS_KEY_RELATIONSHIP]: {
    title: 'AlloyDB Cluster Kms Key Uses Relationship',
    description: 'Build AlloyDB Cluster Kms Key Has Relationship.',
    defaultsToDisabled: false,
  },
};

// IAM Permissions
export const AlloyDBPermissions = {
  STEP_ALLOYDB_POSTGRE_SQL_CLUSTER: ['alloydb.clusters.list'],
  STEP_ALLOYDB_POSTGRE_SQL_INSTANCE: ['alloydb.instances.list'],
  STEP_ALLOYDB_POSTGRE_SQL_CONNECTION: ['alloydb.instances.connect'],
  STEP_ALLOYDB_POSTGRE_SQL_BACKUP: ['alloydb.backups.get'],
};
