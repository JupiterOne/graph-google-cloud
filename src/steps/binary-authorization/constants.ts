export const BINARY_AUTHORIZATION_POLICY_ENTITY_CLASS = 'AccessPolicy';
export const BINARY_AUTHORIZATION_POLICY_ENTITY_TYPE =
  'google_binary_authorization_policy';

export const STEP_BINARY_AUTHORIZATION_POLICY =
  'fetch-binary-authorization-policy';

export const RELATIONSHIP_TYPE_PROJECT_HAS_BINARY_AUTHORIZATION_POLICY =
  'google_cloud_project_has_binary_authorization_policy';

export const IngestionSources = {
  BINARY_AUTHORIZATION_POLICY: 'binary-authorization-policy',
};

export const BinaryAuthorizationIngestionConfig = {
  [IngestionSources.BINARY_AUTHORIZATION_POLICY]: {
    title: 'Google Cloud Binary Authorization Policy',
    description: 'Controls for deploying trusted containers.',
    defaultsToDisabled: false,
  },
};

export const BinaryAuthPermissions = {
  STEP_BINARY_AUTHORIZATION_POLICY: ['binaryauthorization.policy.get'],
};
