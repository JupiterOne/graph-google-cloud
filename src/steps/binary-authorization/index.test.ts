import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { integrationConfig } from '../../../test/config';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchBinaryAuthorizationPolicy } from '.';
import { fetchResourceManagerProject } from '../resource-manager/index';
import {
  BINARY_AUTHORIZATION_POLICY_ENTITY_TYPE,
  RELATIONSHIP_TYPE_PROJECT_HAS_BINARY_AUTHORIZATION_POLICY,
} from './constants';
import { ResourceManagerEntities } from '../resource-manager/constants';

describe('#fetchBinaryAuthorization', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchBinaryAuthorization',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: {
        ...integrationConfig,
        serviceAccountKeyFile: integrationConfig.serviceAccountKeyFile.replace(
          'j1-gc-integration-dev-v2',
          'j1-gc-integration-dev-v3',
        ),
        serviceAccountKeyConfig: {
          ...integrationConfig.serviceAccountKeyConfig,
          project_id: 'j1-gc-integration-dev-v3',
        },
      },
    });

    await fetchResourceManagerProject(context);
    await fetchBinaryAuthorizationPolicy(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === ResourceManagerEntities.PROJECT._type,
      ),
    ).toMatchGraphObjectSchema(ResourceManagerEntities.PROJECT);

    expect(
      context.jobState.collectedEntities.filter(
        (e) => e._type === BINARY_AUTHORIZATION_POLICY_ENTITY_TYPE,
      ),
    ).toMatchGraphObjectSchema({
      _class: ['AccessPolicy'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'google_binary_authorization_policy' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          admissionWhitelistPatterns: {
            type: 'array',
            items: { type: 'string' },
          },
          evaluationMode: { type: 'string' },
          globalPolicyEvaluationMode: { type: 'string' },
          webLink: { type: 'string' },
          updatedOn: { type: 'number' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) =>
          e._type === RELATIONSHIP_TYPE_PROJECT_HAS_BINARY_AUTHORIZATION_POLICY,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'google_cloud_project_has_binary_authorization_policy',
          },
        },
      },
    });
  });
});
