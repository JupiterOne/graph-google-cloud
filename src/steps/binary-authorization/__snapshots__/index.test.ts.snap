// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchBinaryAuthorization should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Account",
      ],
      "_key": "projects/167984947943",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "createTime": "2021-05-31T15:15:56.125Z",
            "displayName": "j1-gc-integration-dev-v3",
            "etag": "6nNZKrLueq5A5QHM92fcDg==",
            "name": "projects/167984947943",
            "parent": "folders/306329820365",
            "projectId": "j1-gc-integration-dev-v3",
            "state": "ACTIVE",
            "updateTime": "2021-05-31T15:15:58.148Z",
          },
        },
      ],
      "_type": "google_cloud_project",
      "createdOn": 1622474156125,
      "displayName": "j1-gc-integration-dev-v3",
      "id": "j1-gc-integration-dev-v3",
      "lifecycleState": "ACTIVE",
      "name": "projects/167984947943",
      "parent": "folders/306329820365",
      "projectId": "j1-gc-integration-dev-v3",
      "updatedOn": 1622474158148,
      "webLink": "https://console.cloud.google.com/home/dashboard?project=j1-gc-integration-dev-v3",
    },
    {
      "_class": [
        "AccessPolicy",
      ],
      "_key": "projects/j1-gc-integration-dev-v3/policy",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "admissionWhitelistPatterns": [
              {
                "namePattern": "gcr.io/google_containers/*",
              },
              {
                "namePattern": "gcr.io/google-containers/*",
              },
              {
                "namePattern": "k8s.gcr.io/*",
              },
              {
                "namePattern": "gke.gcr.io/*",
              },
              {
                "namePattern": "gcr.io/stackdriver-agents/*",
              },
            ],
            "defaultAdmissionRule": {
              "enforcementMode": "ENFORCED_BLOCK_AND_AUDIT_LOG",
              "evaluationMode": "ALWAYS_ALLOW",
            },
            "globalPolicyEvaluationMode": "ENABLE",
            "name": "projects/j1-gc-integration-dev-v3/policy",
          },
        },
      ],
      "_type": "google_binary_authorization_policy",
      "admissionWhitelistPatterns": [
        "gcr.io/google_containers/*",
        "gcr.io/google-containers/*",
        "k8s.gcr.io/*",
        "gke.gcr.io/*",
        "gcr.io/stackdriver-agents/*",
      ],
      "createdOn": undefined,
      "displayName": "projects/j1-gc-integration-dev-v3/policy",
      "evaluationMode": "ALWAYS_ALLOW",
      "globalPolicyEvaluationMode": "ENABLE",
      "name": "projects/j1-gc-integration-dev-v3/policy",
      "updatedOn": undefined,
      "webLink": "https://console.cloud.google.com/security/binary-authorization/policy?project=j1-gc-integration-dev-v3&folder=&organizationId=",
    },
  ],
  "collectedRelationships": [
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/167984947943",
      "_key": "projects/167984947943|has|projects/j1-gc-integration-dev-v3/policy",
      "_toEntityKey": "projects/j1-gc-integration-dev-v3/policy",
      "_type": "google_cloud_project_has_binary_authorization_policy",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": [
    "google_cloud_project",
    "google_binary_authorization_policy",
    "google_cloud_project_has_binary_authorization_policy",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
